common/crypto/src/asymmetric/identity/mod.rs

```diff
@@ -62,16 +62,31 @@ pub struct KeyPair {
     // nothing secret about public key
     #[zeroize(skip)]
     public_key: PublicKey,
+
+    #[zeroize(skip)]
+    index: u32,
 }
 
 impl KeyPair {
     #[cfg(feature = "rand")]
     pub fn new<R: RngCore + CryptoRng>(rng: &mut R) -> Self {
+        let index = rng.gen();
```

Comment:
I'm not sure I understand the point of having a random index for the random keys

Reply:
Its just to have something, that is not None or constant

Reply:
I'll document the intent, otherwise I'll forget :D